packages/swing-store/src/snapStore.js

```diff
@@ -154,72 +189,41 @@ export function makeSnapStore(
           snapReader.destroy();
         });
 
-        // Create a file for the compressed snapshot in-place
-        // to be atomically renamed once we know its name
-        // from the hash of raw snapshot contents.
-        const {
-          path: tmpGzPath,
-          fd: tmpGzFd,
-          cleanup: tmpGzCleanup,
-        } = await ptmpFile({ tmpdir: root });
-        cleanup.push(tmpGzCleanup);
-        const gzWriter = createWriteStream(noPath, {
-          fd: tmpGzFd,
-          autoClose: false,
-        });
-        cleanup.push(() => gzWriter.close());
-
-        await Promise.all([fsStreamReady(snapReader), fsStreamReady(gzWriter)]);
+        await fsStreamReady(snapReader);
 
         const hashStream = createHash('sha256');
         const gzip = createGzip();
+        let compressedByteCount = 0;
 
         const { result: hash, duration: compressSeconds } =
           await measureSeconds(async () => {
             snapReader.pipe(hashStream);
-            snapReader.pipe(gzip).pipe(gzWriter);
-
-            await Promise.all([finished(snapReader), finished(gzWriter)]);
+            const compressedSnapshot = await buffer(snapReader.pipe(gzip));
+            await finished(snapReader);
 
             const h = hashStream.digest('hex');
-            await pfsync(tmpGzFd);
-
-            const tmpGzClose = cleanup.pop();
-            tmpGzClose();
+            sqlStopUsingLastSnapshot.run(vatID);
+            if (!keepSnapshots) {
+              deleteAllUnusedSnapshots();
+            }
+            compressedByteCount = compressedSnapshot.length;
+            sqlSaveSnapshot.run(
+              vatID,
+              endPos,
+              h,
```

Comment:
nit: can we call this `hash` instead of `h`? to match the schema

Reply:
The short answer is no; doing that will trip over the eslint rule forbidding name shadowing.  It's possible some weird rewrite could address that but I'm not sure it would be worth the bother.